### PR TITLE
Improved UiToIdeContentProposalProvider

### PR DESCRIPTION
**Pull request by cdietrich**
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

**Comment by cdietrich** [comment]
please review

File: org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/contentassist/AbstractCompletionProposalFactory.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import org.eclipse.swt.graphics.Image;
+
+import com.google.inject.Inject;
+
+/**
+ * @author Christian Dietrich - Initial contribution and API
+ * 
+ * @since 2.13
+ */
+public abstract class AbstractCompletionProposalFactory implements ICompletionProposalFactory{
```
Comment (kthoms) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/contentassist/AbstractCompletionProposalFactory.java]: please leave a space for {

File: org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/contentassist/UiToIdeContentProposalProvider.xtend [hunk 2]
```diff
@@ -99,22 +100,22 @@ class UiToIdeContentProposalProvider extends AbstractContentProposalProvider {
...
-    }
-    
-    /**
-     * This method does nothing and should not be used.
-     */
-    override final completeRuleCall(RuleCall object, ContentAssistContext context, ICompletionProposalAcceptor acceptor) {
-    }
+	/** 
+	 * Returns the image for the label of the given element.
+	 * @param descriptionthe {@link IEObjectDescription} for which to provide the label image
```
Comment (kthoms) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/contentassist/UiToIdeContentProposalProvider.xtend]: space after `description` missing

File: org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/contentassist/UiToIdeContentProposalProvider.xtend [hunk 3]
```diff
@@ -99,22 +100,22 @@ class UiToIdeContentProposalProvider extends AbstractContentProposalProvider {
...
+	 * @return the image used to label the element, or <code>null</code> if there is no image for the given object
+	 * @noreference This method is not intended to be referenced by clients.
+	 */
+	def protected Image getImage(IEObjectDescription description) {
+		return getImage(description.getEObjectOrProxy()) 
+	}
+	/** 
+	 * Returns the image for the label of the given element.
+	 * @param eObjectthe element for which to provide the label image
```
Comment (kthoms) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/contentassist/UiToIdeContentProposalProvider.xtend]: space after `eObject` missing

File: org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/contentassist/AbstractCompletionProposalFactory.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+		ConfigurableCompletionProposal result = doCreateProposal(proposal, displayString, image, replacementOffset,
+				replacementLength);
+		result.setPriority(priority);
+		result.setMatcher(context.getMatcher());
+		result.setReplaceContextLength(getReplacementContextLength(context));
+		return result;
+	}
+	
+	/**
+	 * @since 2.12
```
Comment (kthoms) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/contentassist/AbstractCompletionProposalFactory.java]: the class is @since 2.13, remove?

**Comment by cdietrich** [comment]
updated @kthoms

**Pull request merged**